src/Controls/src/Core/Platform/GestureManager/GestureManager.Gtk.cs

```diff
@@ -1,17 +1,268 @@
+				{
+					_control.ButtonPressEvent -= OnControlButtonPressEvent;
+				}
+
+				_control = value;
+
+				if (PreventGestureBubbling)
+				{
+					UpdatingGestureRecognizers();
+				}
+			}
+		}
+
+		public VisualElement? Element
+		{
+			get { return _element; }
+			set
+			{
+				if (_element == value)
+					return;
+
+				if (_element != null)
+				{
+					var view = _element as View;
+					if (view != null)
+					{
+						var oldRecognizers = (ObservableCollection<IGestureRecognizer>)view.GestureRecognizers;
+						oldRecognizers.CollectionChanged -= _collectionChangedHandler;
+					}
+				}
+
+				_element = value;
+
+				if (_element != null)
+				{
+					var view = _element as View;
+					if (view != null)
+					{
+						var newRecognizers = (ObservableCollection<IGestureRecognizer>)view.GestureRecognizers;
+						newRecognizers.CollectionChanged += _collectionChangedHandler;
+					}
+				}
+			}
+		}
+
+		public bool PreventGestureBubbling
+		{
+			get
+			{
+				return Element switch
+				{
+					Button => true,
+					CheckBox => true,
+					DatePicker => true,
+					Stepper => true,
+					Slider => true,
+					Switch => true,
+					TimePicker => true,
+					ImageButton => true,
+					RadioButton => true,
+					_ => false,
+				};
+			}
 		}
 
 		public void Dispose()
 		{
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+
+		private void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+				return;
+
+			_isDisposed = true;
+
+			if (!disposing)
+				return;
+
+			if (_container != null)
+			{
+				_container.ButtonPressEvent -= OnContainerButtonPressEvent;
+			}
+
+			if (_element != null)
+			{
+				var view = _element as View;
+				if (view != null)
+				{
+					var oldRecognizers = (ObservableCollection<IGestureRecognizer>)view.GestureRecognizers;
+					oldRecognizers.CollectionChanged -= _collectionChangedHandler;
+				}
+			}
+
+			if (_control != null)
+			{
+				_control.ButtonPressEvent -= OnControlButtonPressEvent;
+			}
+
+			Container?.Destroy();
+			Container = null;
+		}
+
+		private void ModelGestureRecognizersOnCollectionChanged(object? sender, NotifyCollectionChangedEventArgs notifyCollectionChangedEventArgs)
+		{
+			UpdatingGestureRecognizers();
+		}
+
+		private void UpdatingGestureRecognizers()
+		{
+			var view = Element as View;
+			var gestures = view?.GestureRecognizers;
+
+			if (_container == null || gestures == null)
+				return;
+
+			_container.ButtonPressEvent -= OnContainerButtonPressEvent;
+
+			if (gestures.GetGesturesFor<TapGestureRecognizer>().Any() || gestures.GetGesturesFor<ClickGestureRecognizer>().Any())
+			{
+				_container.ButtonPressEvent += OnContainerButtonPressEvent;
+			}
+			else
+			{
+				if (_control != null && PreventGestureBubbling)
+				{
+					_control.ButtonPressEvent += OnControlButtonPressEvent;
+				}
+			}
 		}
+
+		private void OnContainerButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			var button = args.Event.Button;
+			if (button != 1 && button != 3)
```

Comment:
Since 3 means "both primary and secondary buttons", equivalent is `button == ButtonsMask.Secondary`, right?

Reply:
@aarani ^

Reply:
Why do we return if `button == ButtonsMask.Secondary`?

Reply:
I think the logic is that right mouse button clicks are ignored.

Reply:
It doesn't make any sense, what about this then? https://github.com/nblockchain/CrossMaui/pull/14/files#diff-02814246dfde7d24f71dbab5f98e006735cfc149294761778c19cbfd4e7b2942R255

Reply:
are you sure 3 <> ButtonsMask.Secondary?

Reply:
@aarani 
See https://github.com/nblockchain/CrossMaui/blob/master/src/Controls/src/Core/ButtonsMask.cs
ButtonsMask.Secondary is 2

Reply:
The same is in Xamarin: https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Core/ClickGestureRecognizer.cs#L8

Reply:
@aarani can I merge?

Reply:
I'm not sure I understand the logic but since this is a port of https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/VisualElementTracker.cs. I can get behind it

Reply:
Let's keep the != 1 && != 3 except that I'm fine.